frontend/src/app/NotificationContext.jsx

```diff
@@ -0,0 +1,82 @@
+import { createContext, useState, useEffect, useContext } from "react";
+import axios from "axios";
+import { io } from "socket.io-client";
+
+const userInfo = JSON.parse(localStorage.getItem("userInfo"));
+const token = localStorage.getItem("jwt");
+const BACKEND_URL = import.meta.env.VITE_BACKEND_URL;
+const socket = io(BACKEND_URL, {
+  withCredentials: true,
+});
+
+const notifAction = async (socket_name) => {
+  socket.on(socket_name, async (data) => {
+    const notif_data = JSON.stringify(data);
+    try {
+      const response = await axios.post(
+        `${BACKEND_URL}/notif/new/notification`,
+        { data: notif_data },
+        { withCredentials: true }
+      );
+    } catch (error) {
+      setError("Error generating new notification");
+    }
+  });
+};
+notifAction("notify_group_create_file");
```

Comment:
Same comment as the last PR about used a fixed object to represent the set of options

Reply:
> Same comment as the last PR about used a fixed object to represent the set of options

Fixed this in https://github.com/izuchukwumba/peered/pull/9/commits/ba99fb3187dd4f3816abcf7b2aa0c716adbd2961